pep-0XXX-environment-markers.rst

```diff
@@ -0,0 +1,171 @@
+PEP: 0XXX
+Title: Environment Markers
+Version: $Revision$
+Last-Modified: $Date$
+Author: James Polley <jp@jamezpolley.com>
+BDFL-Delegate: Nick Coghlan <ncoghlan@gmail.com>
+Status: Draft
+Type: Informational
+Content-Type: text/x-rst
+Created: 03-Jul-2015
+Post-History: 03-Jul-2015
+
+
+Abstract
+========
+
+An **environment marker** describes a condition about the current execution
+environment. They are used to indicate when certain dependencies are only
+required in particular environments, and to indicate supported platforms
+for distributions with additional constraints beyond the availability of a
+Python runtime.
+
+Environment markers were first specified in PEP-0345[1]. PEP-0426[2] (which
+would replace PEP-0345) proposed extensions to the markers. When
+2.7.10 was released, even these extensions became insufficient due to
+their reliance on simple lexical comparisons, and thus this PEP has
+been born.
+
+
+Rationale
+=========
+
+Many Python packages are written with portability in mind.
+
+For many packages this means they aim to support a wide range of
+Python releases. If they depend on libraries such as ``argparse`` -
+which started as external libraries, but later got incorporated into
+core - specifying a single set of requirements is difficult, as the
+set of required packages differs depending on the version of Python in
+use.
+
+For other packages, designing for portability means supporting
+multiple operating systems. However, the significant differences
+between them may mean that particular dependencies are only needed on
+particular platforms (relying on ``pywin32`` only on Windows, for
+example)"
+
+Environment Markers attempt to provide more flexibility in a list of
+requirements by allowing the developer to list requirements that are
+specific to a particular environment.
+
+Examples
+========
+
+Here are some examples of such markers inside a requirements.txt::
+
+   pywin32 >=1.0 ; sys_platform == 'win32'
+   unittest2 >=2.0,<3.0 ; python_version == '2.4' or python_version == '2.5'
+   backports.ssl_match_hostname >= 3.4 ; python_version < '2.7.9' or (python_version >= '3.0' and python_version < '3.4')
```

Comment:
This could be written more cleanly as:

backports.ssl_match_hostname >= 3.4 ; python_version < '3.4' and python_version != 2.7.10

however that means we don't have a demonstrating of using parens. I think it's probably more important to give a readable example though.